File: packages/js-dpp/lib/errors/InvalidIdentifierError.js
const ConsensusError = require('./ConsensusError');

class InvalidIdentifierError extends ConsensusError {
  /**
   * @param {string} identifierName
   * @param {IdentifierError} error
   */
  constructor(identifierName, error) {
    super(`Invalid ${identifierName}: ${error.message}`);

    this.identifierName = identifierName;
    this.error = error;
  }

  /**
   * Get identifier name
   *
   * @return {string}
   */
  getIdentifierName() {
    return this.identifierName;
  }

  /**
   * Get identifier error
   *
   * @return {IdentifierError}
   */
  getIdentifierError() {
    return this.error;
  }
}

module.exports = InvalidIdentifierError;

File: packages/rs-dpp/src/errors/consensus/basic/invalid_identifier_error.rs
use crate::errors::consensus::basic::BasicError;
use crate::errors::consensus::ConsensusError;
use crate::errors::ProtocolError;
use bincode::{Decode, Encode};
use platform_serialization_derive::{PlatformDeserialize, PlatformSerialize};
use thiserror::Error;

#[derive(
    Error, Debug, Clone, PartialEq, Eq, Encode, Decode, PlatformSerialize, PlatformDeserialize,
)]
#[error("Invalid {identifier_name}: {message}")]
#[platform_serialize(unversioned)]
#[cfg_attr(feature = "apple", ferment_macro::export)]
pub struct InvalidIdentifierError {
    /*

    DO NOT CHANGE ORDER OF FIELDS WITHOUT INTRODUCING OF NEW VERSION

    */
    pub identifier_name: String,
    pub message: String,
}

impl InvalidIdentifierError {
    pub fn new(identifier_name: String, message: String) -> Self {
        Self {
            identifier_name,
            message,
        }
    }

    pub fn identifier_name(&self) -> &str {
        &self.identifier_name
    }

    pub fn message(&self) -> &str {
        &self.message
    }
}

impl From<InvalidIdentifierError> for ConsensusError {
    fn from(err: InvalidIdentifierError) -> Self {
        Self::BasicError(BasicError::InvalidIdentifierError(err))
    }
}
